language/vm/vm-runtime/src/libra_vm.rs

```diff
@@ -557,15 +555,23 @@ pub fn chunk_block_transactions(txns: Vec<Transaction>) -> Vec<TransactionBlock>
                 }
                 blocks.push(TransactionBlock::BlockPrologue(data));
             }
-            Transaction::WriteSet(ws) => {
+            Transaction::WriteSet(cs) => {
                 if !buf.is_empty() {
                     blocks.push(TransactionBlock::UserTransaction(buf));
                     buf = vec![];
                 }
-                blocks.push(TransactionBlock::WriteSet(ws));
+                blocks.push(TransactionBlock::WriteSet(cs));
             }
             Transaction::UserTransaction(txn) => {
-                buf.push(txn);
+                if let TransactionPayload::WriteSet(cs) = txn.payload() {
```

Comment:
why would a `Transaction::UserTransaction` be also a `TransactionPayload::WriteSet` now? or asking another way why does a `WriteSet` come in both `Transaction::WriteSet` and `Transaction::UserTransaction`?

Reply:
Technically `TransactionPayload::WriteSet` should be equivalent to `Transaction::WriteSet`. I was trying to merge the code logic for both.

Reply:
Is there a reason for having TransactionPayload::WriteSet?